error-handling/src/test/java/com/palantir/remoting1/errors/SerializableErrorErrorDecoderTests.java

```diff
@@ -159,16 +163,11 @@ private static void testEncodingAndDecodingWebException(Class<? extends WebAppli
             throw new RuntimeException(e);
         }
 
-        Exception exception = encodeAndDecode(exceptionToProcess);
-        assertThat(exception.getCause(), is(instanceOf(exceptionClass)));
-        assertEquals(status, ((WebApplicationException) exception.getCause()).getResponse()
-                .getStatusInfo());
-        assertEquals(message, exception.getCause().getMessage());
-
-        assertThat(exception, is(instanceOf(exceptionClass)));
-        assertEquals(status, ((WebApplicationException) exception).getResponse()
-                .getStatusInfo());
-        assertEquals(message, exception.getMessage());
+        RemoteException exception = (RemoteException) encodeAndDecode(exceptionToProcess);
+        assertThat(exception.getCause(), is(nullValue()));
```

Comment:
yes.

On Mon, Aug 15, 2016 at 7:42 AM Joe Ellis <notifications@github.com> wrote:

> In
> error-handling/src/test/java/com/palantir/remoting1/errors/SerializableErrorErrorDecoderTests.java
> <https://github.com/palantir/http-remoting/pull/199#discussion_r74771619>:
>
> > @@ -159,16 +163,11 @@ private static void testEncodingAndDecodingWebException(Class<? extends WebAppli
> >              throw new RuntimeException(e);
> >          }
> >
> > -        Exception exception = encodeAndDecode(exceptionToProcess);
> > -        assertThat(exception.getCause(), is(instanceOf(exceptionClass)));
> > -        assertEquals(status, ((WebApplicationException) exception.getCause()).getResponse()
> > -                .getStatusInfo());
> > -        assertEquals(message, exception.getCause().getMessage());
> > -
> > -        assertThat(exception, is(instanceOf(exceptionClass)));
> > -        assertEquals(status, ((WebApplicationException) exception).getResponse()
> > -                .getStatusInfo());
> > -        assertEquals(message, exception.getMessage());
> > +        RemoteException exception = (RemoteException) encodeAndDecode(exceptionToProcess);
> > +        assertThat(exception.getCause(), is(nullValue()));
>
> is the cause of all RemoteExceptions expected to be null?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/199/files/d98499a7bbdd8b9a330df53d6289ac6259c79baa#r74771619>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdwTTuIPsx49PGlBgpRRR1OSZPAU1Lks5qgHrHgaJpZM4Jjywx>
> .
>